sdk/src/transaction.rs

```diff
@@ -246,6 +255,26 @@ impl Transaction {
             .collect())
     }
 
+    /// Replace all the signatures and pubkeys
+    pub fn replace_signatures(&mut self, signers: &[(Pubkey, Signature)]) -> Result<()> {
```

Comment:
How about moving signature verification here instead of inside a separate method? In order to replace a signature it must be verified first? That way `verify` can't be forgotten?